src/Transition/CreateNewIDFUsingRulesV24_2_0.f90

```diff
@@ -125,24 +125,42 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
 
-  ! TODO: Move to V10_0_0.f90 when available
-  ! For Defaulting now-required RunPeriod Name
-  INTEGER :: TotRunPeriods = 0
-  INTEGER :: runPeriodNum = 0
-  INTEGER :: iterateRunPeriod = 0
-  CHARACTER(len=MaxNameLength), ALLOCATABLE, DIMENSION(:) :: CurrentRunPeriodNames
-  CHARACTER(len=20) :: PotentialRunPeriodName
-  ! END OF TODO
-
-  ! used in transition code for HeatExchanger:AirToAir:SensibleAndLatent
-  CHARACTER(20), DIMENSION(4) :: HxEffectAt75Airflow
-  CHARACTER(20), DIMENSION(4) :: HxEffectAt100Airflow
-  CHARACTER(MaxNameLength + 2), DIMENSION(4) :: HxTableName
-  LOGICAL :: tableAdded
-  LOGICAL :: tableIndependentVarAdded = .false.
-  CHARACTER(10) :: tableID
-  REAL :: effect75
-  REAL :: effect100
+  INTEGER :: TotSPMs = 0
+  INTEGER :: spmNum = 0
+  INTEGER :: spmIndex = 0
+  CHARACTER(len=MaxNameLength), ALLOCATABLE, DIMENSION(:) :: SPMNames
+
+  INTEGER, PARAMETER :: NumSPMTypes = 29
+  CHARACTER(len=MaxNameLength), DIMENSION(NumSPMTypes) :: SPMTypes
+  SPMTypes(1) = "SETPOINTMANAGER:SCHEDULED"
```

Comment:
I can't really read Fortran, and by "can't" I mean "don't want to." Why is this thing in Fortran and not in Python? This seems like a very Python thing.

Reply:
You are right that this is a Python thing.  It is actively being worked on by @jasondegraw, and it looks great.  We'll get there.